src/utils/FormattingUtils.ts

```diff
@@ -73,30 +74,11 @@ export function formatBytes(bytes: number, decimals = 2): string {
 export function formatCryptoKey(key: string): string {
     return key.match(/.{1,4}/g)!.join(" ");
 }
-/**
- * calculates a numeric hash for a given string
- *
- * @param {string} str string to hash
- *
- * @return {number}
- */
-export function hashCode(str?: string): number {
-    let hash = 0;
-    let chr: number;
-    if (!str?.length) {
-        return hash;
-    }
-    for (let i = 0; i < str.length; i++) {
-        chr = str.charCodeAt(i);
-        hash = (hash << 5) - hash + chr;
-        hash |= 0;
-    }
-    return Math.abs(hash);
-}
 
-export function getUserNameColorClass(userId?: string): string {
-    const colorNumber = (hashCode(userId) % 8) + 1;
-    return `mx_Username_color${colorNumber}`;
+export function getUserNameColorClass(userId: string): string {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
Can I have a comment explaining why this is OK to disable?

Reply:
Same comment about a test as above.

Reply:
The function starts with `use` but isn't actually a hook. The underlying class is tested within Compound but not here. I might land this to fix the dev build and get back to it

Reply:
Great, can we have that comment in the code please?

Reply:
We should avoid the naming as react will apply hook behaviours to it if called from functional components which adds performance overhead.